unic/char/range/src/range.rs

```diff
@@ -0,0 +1,172 @@
+use std::char;
+use std::collections::Bound;
+use CharIter;
+
+/// A range of unicode code points.
+///
+/// The most idiomatic way to construct this range is through the use of the `chars!` macro:
+///
+/// ```
+/// # #[macro_use] extern crate unic_char_range;
+/// # use unic_char_range::*;
+/// # fn main() {
+/// assert_eq!(chars!('a'..='z'), CharRange::closed('a', 'z'));
+/// assert_eq!(chars!('a'..'z'), CharRange::open_right('a', 'z'));
+/// assert_eq!(chars!(..), CharRange::all());
+/// # }
+/// ```
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Comment:
I actually did a bit of reading on `RangeInclusive` while making this and I _think_ the current design that's been landed on is having just two fields--a start and a stop--and creating `Step` fn to make a known-ended state.

And yes, it makes sense that two empty ranges no matter their internal state should compare the same.